sqlmesh/core/renderer.py

```diff
@@ -166,7 +167,19 @@ def _render(
 
                 if expression:
                     with _normalize_and_quote(expression, self._dialect) as expression:
-                        pass
+                        if hasattr(expression, "selects"):
+                            for select in expression.selects:
+                                if not isinstance(select, exp.Alias) and select.output_name not in (
```

Comment:
Shouldn't we do this check **after** we qualify the query in the `_optimize_query`?

Reply:
it used to be there, but there's no need for it. optimize_query, if successful already does exactly that.